telemetry/prometheus/src/main/scala/io/buoyant/telemetry/prometheus/PrometheusTelemeter.scala

```diff
@@ -75,14 +75,21 @@ class PrometheusTelemeter(metrics: MetricsTree, private[prometheus] val handlerP
     val (prefix1, labels1) = prefix0 match {
       case Seq("rt", router) if !labelExists(labels0, "rt") =>
         (Seq("rt"), labels0 :+ ("rt" -> escapeLabelVal(router)))
-      case Seq("rt", "service", path) if !labelExists(labels0, "service") =>
-        (Seq("rt", "service"), labels0 :+ ("service" -> escapeLabelVal(path)))
-      case Seq("rt", "client", id) if !labelExists(labels0, "client") =>
-        (Seq("rt", "client"), labels0 :+ ("client" -> escapeLabelVal(id)))
+
+      // Add label for peer { "service", "client", "server" }
+      case Seq("rt", peer, identifier) if !labelExists(labels0, peer) =>
+        (Seq("rt", peer), labels0 :+ (peer -> escapeLabelVal(identifier)))
+
+      // Handle client service case
       case Seq("rt", "client", "service", path) if !labelExists(labels0, "service") =>
         (Seq("rt", "client", "service"), labels0 :+ ("service" -> escapeLabelVal(path)))
-      case Seq("rt", "server", srv) if !labelExists(labels0, "server") =>
-        (Seq("rt", "server"), labels0 :+ ("server" -> escapeLabelVal(srv)))
+
+      // Add label for exception { "failures", "exn" }
+      case Seq("rt", peer, "failures", path) if !labelExists(labels0, "exception") =>
+        (Seq("rt", peer, "failures"), labels0 :+ ("exception" -> escapeLabelVal(path)))
+      case Seq("rt", peer, "exn", path) if !labelExists(labels0, "exception") =>
```

Comment:
Will the exception name ever be multiple segments?  e.g. in the case of nested exceptions?  do we need to capture metrics like:
`Seq("rt", "client", "exn", "com.twitter.finagle.SomeParentException", "java.net.SomeChildException")` ?

Reply:
Yes exceptions will be nested, but the stack trace prints out each layer of the nested structure. In the case of an exception that is nested `foo:bar:baz`, it will be printed across three lines like this: `foo`, `foo:bar`, `foo:bar:baz`. The counter will count each layer of the nested structure.